plugin/kubernetes/kubernetes.go

```diff
@@ -274,6 +274,10 @@ func (k *Kubernetes) Records(state request.Request, exact bool) ([]msg.Service,
 		return nil, e
 	}
 
+	if dnsutil.IsReverse(state.Name()) {
```

Comment:
I think this is indeed a better place to put it. Ah yes, we and up here because of the type of the query, instead of in Reverse (as we do for PTR lookups).